src/peft/peft_model.py

```diff
@@ -721,16 +748,82 @@ def load_adapter(self, model_id: str, adapter_name: str, is_trainable: bool = Fa
                     no_split_module_classes=no_split_module_classes,
                     low_zero=(device_map == "balanced_low_0"),
                 )
+
             if isinstance(device_map, str):
                 device_map = infer_auto_device_map(
                     self, max_memory=max_memory, no_split_module_classes=no_split_module_classes
                 )
+
+            prefix = "base_model.model."
+            # rename offload index weight and model names
+            target_modules = self.peft_config["default"].target_modules
+            if offload_index:
+                keys = list(offload_index.keys())
```

Comment:
Since the `load_adapter` method is already quite long, I wonder if we could move the offloading code inside a sub-method. WDYT?